source/math/FloatBox2D.ooc

```diff
@@ -140,25 +140,31 @@ FloatBox2D: cover {
 	adaptTo: func (other: This, weight: Float) -> This {
 		newCenter := FloatPoint2D linearInterpolation(this center, other center, weight)
 		newSize := FloatSize2D linearInterpolation(this size, other size, weight)
-		this createAround(newCenter, newSize)
+		This createAround(newCenter, newSize)
 	}
 	toString: func -> String { "#{this leftTop toString()}, #{this size toString()}" }
 	parse: static func (input: String) -> This {
 		array := input split(',')
 		This new(array[0] toFloat(), array[1] toFloat(), array[2] toFloat(), array[3] toFloat())
 	}
-	create: static func (leftTop: FloatPoint2D, size: FloatSize2D) -> This { This new(leftTop, size) }
-	create: static func ~fromFloats (left, top, width, height: Float) -> This { This new(left, top, width, height) }
-	createAround: static func (center: FloatPoint2D, size: FloatSize2D) -> This { This new(center + (-size) / 2, size) }
+	createAround: static func (center: FloatPoint2D, size: FloatSize2D) -> This { This new(center - (size / 2.0f), size) }
 	bounds: static func (left, right, top, bottom: Float) -> This { This new(left, top, right - left, bottom - top) }
-	bounds: static func ~fromArray (points: FloatPoint2D[]) -> This { This bounds(points as ArrayList<FloatPoint2D>) }
-	bounds: static func ~fromList (points: ArrayList<FloatPoint2D>) -> This {
+	bounds: static func ~fromArray (points: FloatPoint2D[]) -> This {
+		pointsAsVectorList := FloatPoint2DVectorList new()
+		for (i in 0 .. points length)
+			pointsAsVectorList add(points[i])
+		result := This bounds(pointsAsVectorList)
+		pointsAsVectorList free()
```

Comment:
This isn't very good... maybe we can create a new function that takes a `FloatPoint2D*` and a `count: Int` that runs the loop. Then the array version can call that version as `This bounds(points data, points length)` and the `VectorList` version can call `This bounds(points pointer as This*, points count)`.